### DISCUSSION TITLE
SelectMany on Index has missing field values

### DISCUSSION THREAD

**Discussion by harishamdani** [ravendb/ravendb#15271] [Q&A] [answered]
Hi Guys,

I want to ask question about select many on index. I have a legacy code that I have to maintain here, so I would rather not to make any change in the index below. But, if I have to do it, then that's fine. 
 
My problem is, when utilizing this index on the backend, the filter works. However, the output collection had been missing fields when I instantiated them, specifically the "transactionAction" related properties. LocationId, LocationName and TimeLogged in Result object have default value (default DateTime, null, null) instead of the actual ones from Inventory object.

I would expect that this query would simply map the transactionAction into Result object, but apparently it is not. What could possibly went wrong? Or, what should I fix/modify in order to get the transactionAction property values propagated to Result object? 

On the other hand, when I query the Inventory Class, I can still get the whole values in Inventory object, including the TransactionActions, as expected. 
    `
    
        public class Inventory_TimeLogged : AbstractIndexCreationTask<Inventory, Inventory_TimeLogged.Result>
                 {
                      public Inventory_TimeLogged()
                      {
                          Map = inventory => from inventory in inventories
                                 let vendor = LoadDocument<Vendor>(inventory.vendorId)
                                 from transactionAction in inventory.TransactionActions
                                 select new Result
                                 {
                                     Created = vendor.Created,
                                     VendorId = inventory.VendorId,
                                     ClientId = inventory.ClientId,
                                     TimeLogged = transactionAction.Value.TimeLogged,
                                     LocationId = transactionAction.Value.LocationId,
                                     LocationName = transactionAction.Value.LocationName
                                 };                       
                       }
                      
                      public class Result
                      {
                          public string ClientId { get; set; }
                          public string VendorId { get; set; }
                          public DateTime Created { get; set; }
                          public DateTime TimeLogged { get; set; }
                          public string LocationId { get; set; }
                          public string LocationName { get; set; }
                         public TransactionTypeId TransactionTypeId { get; set; }
                      }
                }
                
                public class Inventory 
                {
                    ...
            
                    public Dictionary<TransactionTypes, TransactionAction> TransactionActions { get; set; }
            
                    ...
                }
            
                public enum TransactionTypes
                {
                    Buy,
                    Sell
                }
    
    `
    
    
One solution that I could think of is to modify Inventory.TransactionActions input a simple class instead of dictionary. That way, the mapping is simpler in the index. The new class could have { BuyAction, and SellAction } with the same type. However, the downside of it, it's not completely flat anymore on the index result object. maybe that's why the initial design was using dictionary and it could be flatten during indexing

**Comment by garayx** [comment]
Hi 

Can you send us a small project or unit [test that](https://ravendb.net/docs/article-page/5.4/Csharp/start/test-driver) reproduces the issue?

**Reply by harishamdani** [reply]
@garayx Thanks for the response. We don't have one at the moment but I'll try to create a project with the sample code.

**Comment by johanskoldekrans** [comment, marked as answer]
Have you tried setting this in the index? StoreAllFields(FieldStorage.Yes);

**Reply by ayende** [reply]
That is the right answer. You want to project the results of the index, but you didn't store the fields to support it.

**Reply by harishamdani** [reply]
I see, thanks for the suggestion. I'll try to use it in the index.
